FindMKL.cmake

```diff
@@ -117,114 +183,130 @@ else()
+foreach(_bits "32BIT" "64BIT")
+    set(_mkl_interface_lib ${MKL_INTERFACE_${_bits}_LIB})
+    set(_mkl_blacs_lib ${MKL_BLACS_${_bits}_LIB})
+    set(_mkl_scalapack_lib ${MKL_SCALAPACK_${_bits}_LIB})
+
+    foreach(_threading "SEQ" "OMP")
+        set(_mkl_threading_lib ${MKL_${_threading}_LIB})
+
+        set(_config "${_bits}_${_threading}")
+        string(TOLOWER ${_config} _tgt_config)
+
+        set(_omp_found_var "")
+        set(_omp_target "")
+        if(${_threading} STREQUAL "OMP")
+            set(_omp_found_var "OpenMP_CXX_FOUND")
+            set(_omp_target "OpenMP::OpenMP_CXX")
+        endif()
+
+        list(FIND MKL_FIND_COMPONENTS BLAS_${_config} _blas_cpt_requested)
+        list(FIND MKL_FIND_COMPONENTS BLACS_${_config} _blacs_cpt_requested)
+        list(FIND MKL_FIND_COMPONENTS SCALAPACK_${_config} _scalapack_cpt_requested)
+
+        if(${_blas_cpt_requested} GREATER "-1" OR
+           ${_blacs_cpt_requested} GREATER "-1" OR
+           ${_scalapack_cpt_requested} GREATER "-1")
+            find_package_handle_standard_args(MKL_BLAS_${_config} REQUIRED_VARS _mkl_threading_lib
+                                                                                _mkl_interface_lib
+                                                                                ${_omp_found_var})
+        endif()
+        
+        if(${_blacs_cpt_requested} GREATER "-1" OR
+           ${_scalapack_cpt_requested} GREATER "-1")
+            find_package_handle_standard_args(MKL_BLACS_${_config} REQUIRED_VARS _mkl_blacs_lib
+                                                                                 MKL_BLAS_${_config}_FOUND
+                                                                                 MPI_FOUND)
+        endif()
+
+        if(${_scalapack_cpt_requested} GREATER "-1")
+            find_package_handle_standard_args(MKL_SCALAPACK_${_config} REQUIRED_VARS _mkl_scalapack_lib
+                                                                                     MKL_BLACS_${_config}_FOUND)
+        endif()
+   
+        if(MKL_BLAS_${_config}_FOUND AND NOT TARGET mkl::blas_${_tgt_config})
+            add_library(mkl::blas_${_tgt_config} INTERFACE IMPORTED)
+            set_target_properties(mkl::blas_${_tgt_config} PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${MKL_INCLUDE_DIR}"
+                INTERFACE_LINK_LIBRARIES "${_mkl_interface_lib};${_mkl_threading_lib};${MKL_CORE_LIB};${_omp_target};Threads::Threads")
+        endif()
+
+        if(MKL_BLACS_${_config}_FOUND AND NOT TARGET mkl::blacs_${_tgt_config})
+            add_library(mkl::blacs_${_tgt_config} INTERFACE IMPORTED)
+            set_target_properties(mkl::blacs_${_tgt_config} PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${MKL_INCLUDE_DIR}"
+                INTERFACE_LINK_LIBRARIES "${_mkl_interface_lib};${_mkl_threading_lib};${MKL_CORE_LIB};${_mkl_blacs_lib};${_omp_target};Threads::Threads")
```

Comment:
What do you think about making it more explicit that BLACS depends on the previous one? Like you did for the next SCALAPACK component. I think that what you did for SCALAPACK is more correct and less prone to errors (and avoids duplication).

Something like 
```
INTERFACE_LINK_LIBRARIES "${_mkl_blacs_lib}; mkl::blas_${_tgt_config}"
```

Reply:
Unfortunately, I don't think that's possible here because of ${_mkl_blacs_lib} changes the order of the libraries.